pkg/kudo/operator.go

```diff
@@ -169,19 +174,31 @@ func (builder OperatorBuilder) Do(client client.Client) (Operator, error) {
 // We assume that this is the intended behavior for most test cases.
 // Don't use this for test cases which have multiple Instances for a single OperatorVersion.
 func (operator Operator) Uninstall() error {
+	return operator.UninstallWaitForDeletion(0)
+}
+
+// UninstallWaitForDeletion is the same as Uninstall but
+// initiates a foreground deletion, and waits for the KUDO resources to disappear.
+// Waits up to timeout for the instance to be deleted, and up to 10 seconds for each of OperatorVersion and Operator.
+func (operator Operator) UninstallWaitForDeletion(timeout time.Duration) error {
 	if operator.client.Kudo == nil {
 		return fmt.Errorf("operator is not initialized")
 	}
 
 	options := metav1.DeleteOptions{}
 
+	if timeout != 0 {
+		propagationPolicy := metav1.DeletePropagationForeground
+		options.PropagationPolicy = &propagationPolicy
+	}
+
 	err := operator.client.Kudo.
 		KudoV1beta1().
 		Instances(operator.Instance.Namespace).
 		Delete(operator.Instance.Name, &options)
 	if err != nil {
 		return fmt.Errorf(
-			"failed to delete Instance %s in namespace %s: %w",
+			"failed to delete Instance %s in namespace %s: %v",
```

Comment:
Why is this changed?
The wrapping is on purpose. As this is a library, callers should be able to unwrap the errors here if needed. And as `test-tools` is mostly a thin wrapper around the Kubernetes API, we should expect users to have access to this API as well.

Here and below.

Reply:
Oooh, my IDE complained so I thought that was a typo until I read https://blog.golang.org/go1.13-errors now.
Looks like I need to catch up with the new hotness and upgrade my Intellij :sweat_smile: 
Reverting this part...